tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -271,7 +271,7 @@ static void write_kerntext(void)
 	expect_fault = true;
 	expected_reason = REASON_HW_EXCEPTION;
 	BARRIER();
-	memcpy(&_is_thread_essential, 0, 4);
+	memset(&_is_thread_essential, 0, 4);
```

Comment:
interesting.  so the source address was null which meant it was reading out of 0x0 (which would cause a fault) versus setting it from a constant using memset.

Reply:
Yeah, so, at least not in Nordic legacy platforms ;) But it caused a SecureFault in nRF9160 :p